lib/mix/tasks.ex

```diff
@@ -80,13 +85,29 @@ defmodule Mix.Tasks.Coveralls do
       end
     end)
 
+    sub_dir_set? = "" != common_options[:subdir] |> to_string
+    root_dir_set? = "" != common_options[:rootdir] |> to_string
+    if sub_dir_set? and root_dir_set? do
+      raise ExCoveralls.InvalidOptionError,
+                message: "subdir and rootdir options are exclusive. please specify only one of them."
+    end
     {remaining, common_options}
   end
 
-  defp analyze_sub_apps(options) do
-    type = options[:type] || "local"
-    stats = ExCoveralls.StatServer.get |> MapSet.to_list
-    ExCoveralls.analyze(stats, type, options)
+  def get_stats(stats, options) do
+    sub_dir_set? = "" != options[:subdir] |> to_string
+    root_dir_set? = "" != options[:rootdir] |> to_string
```

Comment:
Maybe extract this piece of code and put it in some private function where it could be used for the entire module and maintain the consistency and reusability of this code...

Reply:
I thought about it, and avoided making a shallow function for now since there are only two places.